webapp-django/crashstats/crashstats/views.py

```diff
@@ -421,53 +392,56 @@ def get_product_versions_for_crashes_per_day(facets, product):
 
-    return [
-        x for x in context['currentversions']
-        if x['release'].lower() in [rel.lower() for rel in nightlies_only]
-    ]
+    versions = {}
+    for product, productversions in context['active_versions'].items():
+        versions[product] = []
+        for version in productversions:
+            rel_release = version['build_type'].lower()
+            if rel_release in [x.lower() for x in nightlies_only]:
+                versions[product].append(version['version'])
+
+    return versions
 
 
 def get_all_nightlies_for_product(context, product):
     nightlies_only = settings.NIGHTLY_RELEASE_TYPES
-
     versions = []
-    for release in context['currentversions']:
-        rel_product = release['product']
-        rel_release = release['release'].lower()
-        if rel_product == product:
-            if rel_release in [x.lower() for x in nightlies_only]:
-                versions.append(release['version'])
+    for version in context['active_versions'].get(product, []):
+        rel_release = version['build_type'].lower()
+        if rel_release in [x.lower() for x in nightlies_only]:
+            versions.append(version['version'])
 
     return versions
 
 
 def get_latest_nightly(context, product):
     version = None
-    for release in context['currentversions']:
-        if release['product'] == product:
-            rel = release['release']
-            if rel.lower() == 'nightly' and release['featured']:
-                version = release['version']
-                break
+    for version in context['active_versions'][product]:
+        rel = version['build_type']
+        if rel.lower() == 'nightly' and version['is_featured']:
+            version = version['version']
+            break
 
     if version is None:
         # We did not find a featured Nightly, let's simply use the latest
-        for release in context['currentversions']:
-            if release['product'] == product:
-                if release['release'].lower() == 'nightly':
-                    version = release['version']
-                    break
+        for version in context['active_versions'].get(product, []):
+            if version['build_type'].lower() == 'nightly':
+                version = version['version']
+                break
 
     return version
 
 
-def get_channel_for_release(version):
-    api = models.CurrentProducts()
-    version_info = api.get(
-        versions=version
-    )
-
-    return version_info['hits'][0]['build_type']
+def get_build_type_for_product_versions(product_versions):
+    """product_versions is a list that looks something like this:
+      ['Firefox:47.0a1']
+    Return the build type on matches
+    """
+    api = models.ProductVersions()
+    for product_version in product_versions:
+        product, version = product_version.split(':')
+        for hit in api.get(product=product, version=version)['hits']:
+            return hit['build_type']
```

Comment:
Don't worry. This ugly thing is part of a the graphs tab (sometimes) of the old reports list. So it's really not important.